### PR TITLE
sort_linter skips on ...length()>1

### LINKED ISSUE #2156: `sort_linter` suggests wrong code when `x[order(...)]` takes multiple vectors
Here is a reprex:
```
lintr::lint(linters = list(lintr::sort_linter()), text = "
x = c(1, 2, 3, 4)
y = c(NA, 10, 20, NA)
x[order(y, x)]
")
#> <text>:4:3: warning: [sort_linter] sort(y, na.last = TRUE) is better than y[order(y, x)].
#> x[order(y, x)]
#>   ^~~~~~~~~~~
```
Obviously `sort(y, na.last = TRUE)` does not generate what `x[order(y, x)]` does.
```
x = c(1, 2, 3, 4)
y = c(NA, 10, 20, NA)
x[order(y, x)]
#> [1] 2 3 1 4
sort(y, na.last = TRUE)
#> [1] 10 20 NA NA
```

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #2156

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -36,7 +36,6 @@
 * `conjunct_test_linter()` also lints usage like `dplyr::filter(x, A & B)` in favor of using `dplyr::filter(x, A, B)` unless `allow_filter = TRUE` (part of #884, @MichaelChirico; #2110, @salim-b).
 * `sort_linter()` checks for code like `x == sort(x)` which is better served by using the function `is.unsorted()` (part of #884, @MichaelChirico).
 * `paste_linter()` gains detection for file paths that are better constructed with `file.path()`, e.g. `paste0(dir, "/", file)` would be better as `file.path(dir, file)` (part of #884, #2082, @MichaelChirico). What exactly gets linted here can be fine-tuned with the `allow_file_path` option (`"double_slash"` by default, with alternatives `"never"` and `"always"`). When `"always"`, these rules are ignored. When `"double_slash"`, paths appearing to construct a URL that have consecutive forward slashes (`/`) are skipped. When `"never"`, even URLs should be construced with `file.path()`.
-* `paste_linter()` gains detection for file paths that are better constructed with `file.path()`, e.g. `paste0(dir, "/", file)` would be better as `file.path(dir, file)` (part of #884, @MichaelChirico).
```
Comment (AshesITR) [inline NEWS.md:39]: ??

File: R/sort_linter.R [hunk 2]
```diff
@@ -69,18 +69,18 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 sort_linter <- function() {
-  order_xpath <- "
+  non_keyword_arg <- "expr[not(preceding-sibling::*[1][self::EQ_SUB])]"
```
Comment (AshesITR) [inline R/sort_linter.R:72]: aka `positional_arg`?

File: NEWS.md (hunk 1 above)
Comment (MichaelChirico) [reply NEWS.md:39]: Bad merge... the bullet before is a duplicate with more info

Comment (MichaelChirico) [reply NEWS.md:39]: Missed here 3592898f656c3a8561e18c9250e3a3a71a034801

File: R/sort_linter.R (hunk 2 above)
Comment (MichaelChirico) [reply R/sort_linter.R:72]: Yes... I do think `non_keyword` maps closer to the XPath using `not()` and referencing `EQ_SUB` though

Comment (AshesITR) [reply R/sort_linter.R:72]: okay

**Review by AshesITR — APPROVED** [review]

**Review by AshesITR — APPROVED** [review]

**Pull request merged**